spawnpoint.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 Siôn le Roux.  All rights reserved.
+// Use of this source code is subject to an MIT-style
+// licence which can be found in the LICENSE file.
+
+package main
+
+import (
+	"math"
+	"math/rand"
+
+	"github.com/solarlune/resolv"
+)
+
+// spawnDistance is the distance where the point is activated
+const spawnDistance = 250
+
+// SpawnPoints is an array of SpawnPoint
+type SpawnPoints []*SpawnPoint
+
+// Update updates all the SpawnPoints
+func (sps *SpawnPoints) Update(g *Game) {
+	for _, s := range *sps {
+		s.Update(g)
+	}
+}
+
+// SpawnPosition describes the spawning position related to the SpawnPoint center
+type SpawnPosition struct {
+	Distance int // Distance of the position from the center
+	Angle    int // Angle of the position
+}
+
+// SpawnPoint is a point on the map where zombies are spawn
+type SpawnPoint struct {
+	Position       Coord
+	InitialCount   int
+	Continuous     bool
+	Zombies        Zombies
+	InitialSpawned bool
+	PrevPosition   SpawnPosition
+	NextSpawn      int
+}
+
+// NextPosition gives the offset of the next spawning to the center of the point
+func (s *SpawnPoint) NextPosition() Coord {
```

Comment:
Cool idea! If a spawn point makes 12 zombies and they all die and the player moves offscreen and it starts spawning again, will it spawn at position #13 or start again from the 0th inner position nearest to the spawn point?

Reply:
I changed it to have restart after 12th position

Reply:
Hmmm, not sure what you mean :)

Reply:
Yes, it uses only 12 distinct positions, then it starts again from position 0 with the next zombie

Reply:
Hmmm ok, I just used 12 as an example, but I meant like "how many it has done so far", e.g. if one spawn point is set to initial=2 and continuous=true then after the first 2 die does it make the next ones further out at position 3 & 4 or at the same place as the first 1 and 2 were?

Reply:
I actually did it to use 12 points :)
YES: "if one spawn point is set to initial=2 and continuous=true then after the first 2 die does it make the next ones further out at position 3 & 4"
And when it reaches 12, then it restarts from 1.
Do you think it is better to use only the initial positions and not all 12?